exporter/exporterhelper/common.go

```diff
@@ -139,12 +140,59 @@ func WithRetry(retrySettings RetrySettings) Option {
 func WithQueue(config QueueSettings) Option {
 	return func(o *baseSettings) {
 		if o.requestExporter {
-			panic("queueing is not available for the new request exporters yet")
+			panic("this option is not available for the new request exporters, " +
+				"use WithMemoryQueue or WithPersistentQueue instead")
 		}
 		o.queueSettings.config = config
 	}
 }
 
+// WithMemoryQueue overrides the default QueueConfig for an exporter to use an in-memory queue.
+func WithMemoryQueue(config QueueConfig) Option {
+	return func(o *baseSettings) {
+		o.queueSettings.config = QueueSettings{
+			Enabled:      config.Enabled,
+			NumConsumers: config.NumConsumers,
+			QueueSize:    config.QueueSize,
+		}
+	}
+}
+
+// WithPersistentQueue overrides the default QueueConfig for an exporter to use a persistent queue.
+// This option can be used only with the new exporter helpers New[Traces|Metrics|Logs]RequestExporter.
+func WithPersistentQueue(config PersistentQueueConfig, marshaler RequestMarshaler, unmarshaler RequestUnmarshaler) Option {
```

Comment:
Or `func WithQueue(config *QueueOption) Option`

Reply:
Alternative for even more generic, you have `func WithQueue(queue Queue) Option`, where Queue is the current ProducerConsumerQueue. This way we can have the queue implementation outside of this package.

Reply:
I like `func WithQueue(queue Queue) Option`. It makes future improvements/additions to the queues easier. But it requires some refactoring because the persistent queue is initialized in two steps currently. I'm trying this out. It has to be refactored anyway I believe

Reply:
@bogdandrutu I submitted a draft PR with the refactoring allowing the `WithQueue` option https://github.com/open-telemetry/opentelemetry-collector/pull/8275. PTAL once you have a chance

Reply:
@bogdandrutu, following the discussion in https://github.com/open-telemetry/opentelemetry-collector/pull/8284#discussion_r1307634509, I'm now more inclined towards this PR other than https://github.com/open-telemetry/opentelemetry-collector/pull/8275 due to the following reasons:

- Exposing `Queue` interface with the `Start` method is fine, but `Callback` and other arguments outlined in https://github.com/open-telemetry/opentelemetry-collector/pull/8284#discussion_r1307634509 cannot be part of the constructors exposed to the users. So, if we don't want them passed as `Start` arguments, we need to encapsulate the internal Queue and hide these arguments. Probably it'll look more like another option that you suggesdted `func WithQueue(config *QueueOption) Option`

- Using the interface from this PR seems like requires the least amount of code to write for the user.

Do you think `WithQueue` is also better in case we might want to introduce another type of Queue in future?

Reply:
The big downside for this is that this package must depend on the queue implementations so if we need to extend in the future we need to grow this package and its dependencies (if extension needs for example S3 queue). Maybe directly Queue is hard to expose, but we can have WithQueue(factory QueueFactory) where the factory gets the Callback as part of the initialization settings.

Reply:
@bogdandrutu I see your point. I took your recommendations from this thread and https://github.com/open-telemetry/opentelemetry-collector/pull/8435 into account and implemented https://github.com/open-telemetry/opentelemetry-collector/pull/8275. I had to introduce a few more packages to avoid circular dependencies. PTAL and let me know if that is close to how you envision it.